bundles/binding/org.openhab.binding.mochadx10/src/main/java/org/openhab/binding/mochadx10/internal/MochadX10Binding.java

```diff
@@ -0,0 +1,204 @@
+	 * 
+	 * @param itemName
+	 * @return The configuration, null otherwise.
+	 */
+	private MochadX10BindingConfig getConfigForItemName(String itemName) {
+		for (MochadX10BindingProvider provider : this.providers) {
+			if (provider.getItemConfig(itemName) != null) {
+				return provider.getItemConfig(itemName);
+			}
+		}
+		logger.warn("No configuration found for item '" + itemName + "'");
+		
+		return null;
+	}
+	
+	@Override
+	protected void internalReceiveCommand(String itemName, Command command) {
+		MochadX10BindingConfig deviceConfig = getConfigForItemName(itemName);
+
+		if (deviceConfig == null) {
+			return;
+		}
+		
+		String address    = deviceConfig.getAddress();
+		String tm         = deviceConfig.getTransmitMethod();
+		String commandStr = "none";
+		Command previousCommand = lastIssuedCommand.get( address );
+		int nrLinesToRead = 2;
+
+		if ( command instanceof OnOffType ) {
+			commandStr = OnOffType.ON.equals( command ) ? "on" : "off";
+		}
+		else if ( command instanceof UpDownType ) {
+			commandStr = UpDownType.UP.equals( command ) ? "bright" : "dim";
+		}
+		else if ( command instanceof StopMoveType ) {
+			if ( StopMoveType.STOP.equals( command ) ) {
+				commandStr = UpDownType.UP.equals( previousCommand ) ? "dim" : "bright";
+			}
+			else {
+				// Move not supported yet
+				commandStr = "none";
+			}
+		}
+		else if ( command instanceof PercentType ) {
+			if ( deviceConfig.getItem() instanceof DimmerItem ) {
+				if ( ((PercentType) command).intValue() == 0 ) {
+					// If percent value equals 0 the x10 "off" command is used instead of xdim
+					commandStr = "off";
+				}
+				else {
+					// 100% maps to value 63 so we need to do scaling
+					long dim_value = Math.round(((PercentType) command).doubleValue() * 63.0/100);
+					commandStr = "xdim " + dim_value;
+				}
+			}
+			else if ( deviceConfig.getItem() instanceof RollershutterItem ) {
+				Double invert_level = 100 - ((PercentType) command).doubleValue();
+				long level = Math.round(invert_level * 25.0/100);
+				commandStr = "extended_code_1 0 1 " + level;
+			}
+			nrLinesToRead = 1;
+		}
+		else if ( command instanceof IncreaseDecreaseType ) {
+			// Increase decrease not yet supported
+			commandStr = "none";
+		}
+
+		try {
+			if ( !commandStr.equals( "none" ) ) {
+				Socket client = new Socket( hostIp, hostPort );
+				client.setSoTimeout(3000);
+				
+				BufferedReader br = new BufferedReader( new InputStreamReader( client.getInputStream() ) );				
+				DataOutputStream out = new DataOutputStream( client.getOutputStream() );
+
+				out.writeBytes(tm + " " + address + " " + commandStr + "\n");
+				logger.debug(tm + " " + address + " " + commandStr);
+				out.flush();
+
+				for (int i = 0; i < nrLinesToRead; i++) {
+					logger.debug( br.readLine() );
+				}
+				
+				out.close();
+				client.close();
+			}
+		} catch (IOException e) {
+			logger.error(e.getMessage());
+		}
+		lastIssuedCommand.put(address, command);
+	}
+}
```

Comment:
every java class should end with an empty line, so please add a new line after the closing bracket